<?xml version="1.0"?>
<robot name="lever_basic">
  <gazebo reference="world">
  </gazebo>
  <gazebo reference="base_link_casing">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.0</mu1>
    <mu2>0.0</mu2>
    <maxVel>1.0</maxVel>
    <minDepth>0.001</minDepth>
    <material>Gazebo/LightGray</material>
  </gazebo>
  <gazebo reference="base_link_squares">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.0</mu1>
    <mu2>0.0</mu2>
    <maxVel>1.0</maxVel>
    <minDepth>0.001</minDepth>
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="lever_link">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>30000.0</mu1>
    <mu2>30000.0</mu2>
    <maxVel>1.0</maxVel>
    <minDepth>0.001</minDepth>
    <material>Gazebo/Black</material>
  </gazebo>
  <material name="light_gray">
    <color rgba="0.9 0.9 0.9 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material> 
  <link name="world">
  </link>
  <link name="base_link_casing">
    <visual>
        <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/base-casing.dae" scale="0.059 0.059 0.059"/>
        </geometry>
        <material name="light_gray"/>
    </visual>
    <collision>
      <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/base-casing.dae" scale="0.059 0.059 0.059"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0.2" rpy="0 0 0"/>
      <mass value="100.0"/>
      <inertia
        ixx="100.0" ixy="0.0" ixz="0.0"
        iyy="100.0" iyz="0.0"
        izz="100.0"/>
    </inertial>
  </link>
  <link name="base_link_squares">
    <contact>
      <lateral_friction value="1.0"/>
      <inertia_scaling value="3.0"/>
    </contact>
    <visual>
        <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/base-squares-only.dae" scale="0.059 0.059 0.059"/>
        </geometry>
        <material name="black"/>
    </visual>
    <collision>
      <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/base-squares-only.dae" scale="0.059 0.059 0.059"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0.2" rpy="0 0 0"/>
      <mass value="1.0"/>
      <inertia
        ixx="1.0" ixy="0.0" ixz="0.0"
        iyy="1.0" iyz="0.0"
        izz="1.0"/>
    </inertial>
  </link>
  <link name="lever_link">
    <contact>
      <lateral_friction value="1.0"/>
      <rolling_friction value="0.0"/>
      <stiffness value="30000"/>
      <damping value="1000"/>
    </contact>
    <visual>
        <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/lever-arm-colored.dae" scale="0.059 0.070 0.059"/>
        </geometry>
        <material name="black"/>
    </visual>
    <collision>
      <geometry>
          <mesh filename="/home/simon/catkin_ws/src/open_manipulator_rl_environments/meshes/lever-arm-colored.dae" scale="0.059 0.070 0.059"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>5.0</mu>
            <mu2>5.0</mu2>
          </ode>
        </friction>
      </surface>
    </collision>
    <inertial>
      <origin xyz="0 0 0.0" rpy="0 0 0"/>
      <mass value="0.1"/>
      <inertia
        ixx="1e-3" ixy="0.0" ixz="0.0"
        iyy="1e-3" iyz="0.0"
        izz="1e-3"/>
    </inertial>
    <contact_coefficients>
      <mu1 value="1.5"/>
      <mu2 value="1.5"/>
      <kp value="0.5"/>
      <kd value="0.5"/>
    </contact_coefficients>
  </link>
  <!-- Can possibly add an offset to this to account for that the origin of the rest is not the bottom of lever-->
   <joint name="world_fixed" type="prismatic">
    <origin rpy="0 0.0 0" xyz="0 0 0.015"/>
    <axis xyz="1 0 0"/>
    <parent link="world"/>
    <child link="base_link_casing"/>
    <dynamics damping="0.0" friction="10000000.0"/>
    <limit effort="10.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>
  <joint name="casing_to_squares" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.0"/>
    <parent link="base_link_casing"/>
  <child link="base_link_squares"/>
  </joint>
  <joint name="joint1" type="revolute">
    <origin rpy="0 0.0 0" xyz="0 0 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-1.57" upper="1.57" velocity="1.0"/>
    <parent link="base_link_squares"/>
    <child link="lever_link"/>
    <dynamics damping="0.0" friction="0.1"/>
  </joint>
  <transmission name="tran1">
    <type>transmission_interface/ActuatorToJointStateInterface</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>
  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/lever_basic</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
      <robotParam>/lever_basic/robot_description</robotParam>
      <legacyModeNS>true</legacyModeNS>
    </plugin>

    <plugin name="p3d_base_controller" filename="libgazebo_ros_p3d.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>100.0</updateRate>
      <bodyName>base_link_casing</bodyName>
      <topicName>lever_basic/odometry</topicName>
      <gaussianNoise>0.00</gaussianNoise>
      <frameName>world</frameName>
      <xyzOffsets>0 0 0</xyzOffsets>
      <rpyOffsets>0 0 0</rpyOffsets>
    </plugin>
  </gazebo>
</robot>